static/app/components/core/segmentedControl/index.chonk.tsx

```diff
@@ -0,0 +1,102 @@
+import type {DO_NOT_USE_ChonkTheme} from '@emotion/react';
+
+import {getChonkButtonStyles} from 'sentry/components/core/button/index.chonk';
+import type {FormSize} from 'sentry/utils/theme';
+import {chonkStyled} from 'sentry/utils/theme/theme.chonk';
+
+export type Priority = 'default' | 'primary';
+
+export const ChonkStyledGroupWrap = chonkStyled('div')<{
+  priority: Priority;
+  size: FormSize;
+}>`
+  position: relative;
+  display: inline-grid;
+  grid-auto-flow: column;
+  min-width: 0;
+
+  ${p => p.theme.form[p.size]}
+
+  & > label:first-child {
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
+  }
+
+  & > label:not(:first-child):not(:last-child) {
+    border-radius: 0;
+
+    &::after {
+      border-left: none;
+      border-right: none;
+    }
+  }
+
+  & > label:last-child {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+  }
+`;
+
+export const ChonkStyledSegmentWrap = chonkStyled('label')<{
+  isSelected: boolean;
+  priority: Priority;
+  size: FormSize;
+  isDisabled?: boolean;
+}>`
+  position: relative;
+  display: flex;
+  align-items: center;
+  margin: 0;
+  cursor: ${p => (p.isDisabled ? 'default' : 'pointer')};
+  min-height: 0;
+  min-width: 0;
+
+  ${p => p.theme.buttonPadding[p.size]}
+  font-weight: ${p => p.theme.fontWeightNormal};
+
+  ${p => ({...getChonkButtonStyles({...p, disabled: p.isDisabled, priority: p.isSelected && p.priority === 'primary' ? 'primary' : 'default'})})}
```

Comment:
took me a while to figure out why `disabled` styles don’t work. `disabled` vs `isDisabled`, and no typescript errors because the prop is optional and there is no excess property checking 😭

Reply:
Oh my god... we really need to delete these dual props, this is a waste of time + having isDisabled and disabled is just confusing